astropy/io/fits/tests/test_hdulist.py

```diff
@@ -376,6 +376,15 @@ def test_file_like_3(self):
         info = [(0, 'PRIMARY', 1, 'PrimaryHDU', 5, (100,), 'int32', '')]
         assert fits.info(self.temp('tmpfile.fits'), output=False) == info
 
+    def test_copy_list(self):
+        """
+        Tests that HDUList.copy() returns an HDUList (regression test
+        for ticket:7211).
+        """
+        hdul = fits.HDUList()
+        new_hdul = hdul.copy()
+        assert isinstance(new_hdul, fits.HDUList)
```

Comment:
Should also test that the *content* is unchanged.

Reply:
It would also be useful to do `assert new_hdul is not hdul` to make sure that we've actually gotten a copy.

Reply:
@pllim Should we check that against an `HDUList` that actually contains `HDU` or just with an empty `HDUList`?

Reply:
Your HDUList should contain HDUs (Primary + Image or BinTable) and check that it is (not) copied depending on the decision on the shallow copy.

Reply:
Alright, I've updated the test. How does it look now?